documents/entity/service_test.go

```diff
@@ -0,0 +1,362 @@
+	assert.Contains(t, err.Error(), "failed to decode identifier")
+	assert.Nil(t, doc)
+
+	// missing last version
+	id := utils.RandomSlice(32)
+	payload.Identifier = hexutil.Encode(id)
+	doc, err = eSrv.DeriveFromUpdatePayload(contextHeader, payload)
+	assert.Error(t, err)
+	assert.True(t, errors.IsOfType(documents.ErrDocumentNotFound, err))
+	assert.Nil(t, doc)
+
+	// failed to load from data
+	old, _ := createCDWithEmbeddedEntity(t)
+	err = testRepo().Create(accountID, old.CurrentVersion(), old)
+	assert.Nil(t, err)
+	payload.Data = &cliententitypb.EntityData{
+		LegalName: "test company",
+		Contacts:  []*entitypb.Contact{{Name: "Mr. Test"}},
+	}
+
+	payload.Identifier = hexutil.Encode(old.ID())
+	doc, err = eSrv.DeriveFromUpdatePayload(contextHeader, payload)
+
+	assert.Error(t, err, "should fail because Identity is missing")
+	assert.Nil(t, doc)
+
+	// failed core document new version
+	payload.Data.LegalName = "new company name"
+	payload.Collaborators = []string{"some wrong ID"}
+	payload.Data.Identity = testingidentity.GenerateRandomDID().String()
+	doc, err = eSrv.DeriveFromUpdatePayload(contextHeader, payload)
+	assert.Error(t, err)
+	assert.True(t, errors.IsOfType(documents.ErrDocumentPrepareCoreDocument, err))
+	assert.Nil(t, doc)
+
+	// success
+	wantCollab := testingidentity.GenerateRandomDID()
+
+	payload.Collaborators = []string{wantCollab.String()}
+	doc, err = eSrv.DeriveFromUpdatePayload(contextHeader, payload)
+	assert.Nil(t, err)
+	assert.NotNil(t, doc)
+	cs, err := doc.GetCollaborators()
+	assert.Len(t, cs, 3)
+	assert.Contains(t, cs, wantCollab)
+	assert.Equal(t, old.ID(), doc.ID())
+	assert.Equal(t, payload.Identifier, hexutil.Encode(doc.ID()))
+	assert.Equal(t, old.CurrentVersion(), doc.PreviousVersion())
+	assert.Equal(t, old.NextVersion(), doc.CurrentVersion())
+	assert.NotNil(t, doc.NextVersion())
+	assert.Equal(t, payload.Data, doc.(*Entity).getClientData())
+}
+
+func TestService_DeriveFromCreatePayload(t *testing.T) {
+	eSrv := service{}
+	ctxh := testingconfig.CreateAccountContext(t, cfg)
+
+	// nil payload
+	m, err := eSrv.DeriveFromCreatePayload(ctxh, nil)
+	assert.Nil(t, m)
+	assert.Error(t, err)
+	assert.True(t, errors.IsOfType(documents.ErrDocumentNil, err))
+
+	// nil data payload
+	m, err = eSrv.DeriveFromCreatePayload(ctxh, &cliententitypb.EntityCreatePayload{})
+	assert.Nil(t, m)
+	assert.Error(t, err)
+	assert.True(t, errors.IsOfType(documents.ErrDocumentNil, err))
+
+	// Init fails
+	payload := &cliententitypb.EntityCreatePayload{
+		Data: &cliententitypb.EntityData{
+			Identity:  "i am not a did",
+			LegalName: "company test",
+		},
+	}
+
+	m, err = eSrv.DeriveFromCreatePayload(ctxh, payload)
+	assert.Nil(t, m)
+	assert.Error(t, err)
+	assert.True(t, errors.IsOfType(documents.ErrDocumentInvalid, err))
+
+	// success
+	payload.Data.Identity = testingidentity.GenerateRandomDID().String()
+	m, err = eSrv.DeriveFromCreatePayload(ctxh, payload)
+	assert.Nil(t, err)
```

Comment:
`assert.NoError(t, err)`
I see other places in entity package where this is repeated, please fix.